rpc/QueryIterator.java

```diff
@@ -48,6 +51,10 @@
 
     @Override
     protected T computeNext() {
+        if (currIterator != null && currIterator.hasNext()) {
```

Comment:
Here we need to exhaust all of the answers in the current message before proceed to the next message.